readium/shared/src/main/java/org/readium/r2/shared/resource/MediaTypeExt.kt

```diff
@@ -21,6 +22,18 @@ public class ContainerMediaTypeSnifferContent(
 
     override suspend fun read(path: String, range: LongRange?): ByteArray? =
         Url.fromDecodedPath(path)?.let { url ->
-            container.get(url).read(range).getOrNull()
+            container.get(url).safeRead(range)
         }
 }
+
+private suspend fun Resource.safeRead(range: LongRange?): ByteArray? {
+    try {
+        // We only read files smaller than 5MB to avoid an [OutOfMemoryError].
+        if (range == null && length().getOrDefault(0) > 5 * 1000 * 1000) {
```

Comment:
We have a `Resource.Exception.OutOfMemory` so it looks like we're supposed to catch `OutOfMemoryError` in `Resource.read` implementations. Anyway, why do we need this trick rather than catching errors?

Reply:
After discussion, we will:
* Remove `Resource.Exception.OutOfMemory` and add a dedicated API to report the OOM issue with PDFium and LCPDF.
* Investigate an alternative to the magic constant of 5MB. For example checking that the file is really a JSON or XML before parsing it, verifying the BOM UTF-8, etc.

In the meantime I'm merging this one to address the crash.